plugin/bookmark.vim

```diff
@@ -262,13 +264,17 @@ function! BookmarkModify(diff)
 
   let current_line = line('.')
   if bm#has_bookmark_at_line(file, current_line)
-    let new_line_nr = current_line + a:diff
+    let new_line_nr = is_abs ? a:diff : current_line + a:diff
+    let last_line_nr = line('$')
+
+    if new_line_nr <= 0 || new_line_nr > last_line_nr
```

Comment:
This is independent from the other changes as it was possible to `:BookmarkMoveUp`/`Down` before line 1 / beyond the last line, making the bookmark disappear.